test/new-e2e/pkg/runner/ci_profile.go

```diff
@@ -38,13 +38,14 @@ func NewCIProfile() (Profile, error) {
 	if err != nil {
 		return nil, fmt.Errorf("unable to get pulumi state password, err: %w", err)
 	}
+	// TODO move to job script
```

Comment:
Why? The goal of profiles + `secretStore` is to do exactly that.

Reply:
I'm triggered when I see usage of `Setenv` in code as it is not thread-safe.

Why are we setting this variable here vs in the CI job definition? We should rather pass it to https://github.com/DataDog/datadog-agent/blob/pducolin/test-infra-move-to-file/test/new-e2e/pkg/utils/infra/stack_manager.go#L367

Reply:
A lot of Go code is not thread safe (all maps), and has no reason to be as only 1 goroutine will ever run it at the same time, which is the case here.

I'm not sure it works setting it inside the workspace, but if it does, it's indeed better to put it there.

Reply:
From the [documentation](https://pkg.go.dev/github.com/pulumi/pulumi/sdk/v2/go/x/auto#EnvVars) it sounds like it does work, forgot to link it to my previous post.

Maps are expected to be not thread safe, `Setenv` like file access is rather unexpected, unless you are familiar with C implementation. So I avoid `Setenv` usage as it is considered a bad practice.